sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs

```diff
@@ -1386,16 +1402,15 @@ private void SetNameFieldsIfNull()
                     $"{nameof(Uri)}: {Uri}");
                 try
                 {
-                    DeleteSnapshotsOptionType? deleteSnapshotsOption;
-
                     // DeleteSnapshotsOptionType.Include is not valid when deleting a Snapshot Share.
                     if (Uri.GetQueryParameters().ContainsKey(Constants.ShareSnapshotParameterName))
                     {
-                        deleteSnapshotsOption = null;
+                        shareSnapshotsDeleteOption = null;
                     }
-                    else
+                    // This is for backwards compatibility.  Perviously, ShareClient.Delete() took a bool includSnapshots parameter.s
+                    else if ((includeSnapshots == null || includeSnapshots == true) && shareSnapshotsDeleteOption == null)
                     {
-                        deleteSnapshotsOption = includeSnapshots ? DeleteSnapshotsOptionType.Include : (DeleteSnapshotsOptionType?)null;
+                        shareSnapshotsDeleteOption = ShareSnapshotsDeleteOption.IncludeLeased;
```

Comment:
Previously, we defaulted to deleting Snapshots with the Share, unless the user specified otherwise.  If a share had snapshots, the call would fail if x-ms-delete-snapshots wasn't set to "include".

Now, it's possible to take a lease on a Share Snapshot.  The delete call will fail if a Share has a Snapshot with a lease, unless x-ms-delete-snapshots = "include-leased"

Reply:
hmm. is leasing snapshot introduced in the same service release or earlier ?

Reply:
It was introduced in the newest service release.

Reply:
so what happens today with leased share snapshots on default parameters? does it fail?

taking that aside, I think that taking lease on snapshot means "please don't touch it" so if someone else is attempting to delete it they should get a warning and if they still want to break/override lease that should be explicit action on their side (i.e. they should specify explicit option to do that.).

Reply:
Previously, you couldn't lease shares or share snapshots.  If you attempted to delete a share that had snapshots, the request would fail with a 400.  The previous behavior in the SDK was to delete snapshots by default.

Today, the previous behavior stands, but has been extended.  Now, you can take a lease on a share snapshot.  If you don't specify "IncludeLeased", the request will fail with a 400.  To keep the previous SDK behavior, I think we need to switch to sending "IncludeLeased" by default.  Passing "Include" by default would also make sense.

Reply:
Passing "Include" by default as we previously did might be the best option.

Reply:
So people who never took lease on snapshot won't tell a difference between default "Include" and "IncludeLeased.
But if we default to "IncludeLeased" we might get some unhappy users who's data got deleted unintentionally.
Afaik for other API/use cases where lease is involved user is required to call some specific action/parameter to break the lease - I think that's more important than making sure that this api never throws 400 on defaults.

I'd default to "Include"

Reply:
> I think that's more important than making sure that this api never throws 400 on defaults.

The one downside of defaulting to "include" is we will throw a 400 with defaults if there are any leased share snapshots associated with the share.

Reply:
I think that's desired. Otherwise why someone would take lease?

Reply:
Changed back to "include"